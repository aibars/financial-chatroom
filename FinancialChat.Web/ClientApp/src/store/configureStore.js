import { applyMiddleware, combineReducers, compose, createStore } from 'redux';
import thunk from 'redux-thunk';
import { connectRouter, routerMiddleware } from 'connected-react-router';
import { authentication } from '.';

export default function configureStore(history, initialState) {
    const middleware = [
        thunk,
        routerMiddleware(history)
    ];

    //console.log('Reducers: ', ...authentication);
    const rootReducer = combineReducers({
        authentication,
        router: connectRouter(history)
    });

    const enhancers = [];
    const windowIfDefined = typeof window === 'undefined' ? null : window;
    if (windowIfDefined && windowIfDefined.__REDUX_DEVTOOLS_EXTENSION__) {
        enhancers.push(windowIfDefined.__REDUX_DEVTOOLS_EXTENSION__());
    }

    return createStore(
        rootReducer,
        initialState,
        compose(applyMiddleware(...middleware), ...enhancers)
    );
}
