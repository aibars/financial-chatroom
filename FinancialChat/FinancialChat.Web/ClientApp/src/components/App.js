import * as React from 'react';
import { Router, Route, Redirect, Switch } from 'react-router';
import Home from './Home';
import PrivateRoute from 'react-private-route';
import { Login } from './Login';
import { history } from '../history';
import '../styles/custom.css'



class App extends React.Component {
    isLoggedIn() {
        return localStorage.user ? true : false;
    }

    render() {
        return (
            <Router history={history}>
                <Switch>
                    <PrivateRoute exact path="/" component={Home}
                        isAuthenticated={!!this.isLoggedIn()} />
                    <Route path="/login" component={Login} />
                    <Redirect from="*" to="/" />
                </Switch>
            </Router>
        );
    }

}

export default App;