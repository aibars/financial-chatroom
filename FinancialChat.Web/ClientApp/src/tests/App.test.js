import React from 'react';
import renderer from 'react-test-renderer';
import App from '../components/App';
import { ConnectedRouter } from 'connected-react-router';
import { history } from '../history';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

const mockStore = configureStore([]);

describe('App', () => {
  let store;
  let component;

  beforeEach(() => {
    store = mockStore({

    });

    store.dispatch = jest.fn();

    component = renderer.create(
      <Provider store={store}>
        <ConnectedRouter history={history}>
          <App />
        </ConnectedRouter>
      </Provider>
    );
  });

  it('should render without crashing', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
});