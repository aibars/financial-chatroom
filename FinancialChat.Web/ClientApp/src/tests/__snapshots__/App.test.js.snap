// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render without crashing 1`] = `
<div
  className="box-container"
>
  <div
    className="inner-container"
  >
    <div
      className="header"
    >
      Login
    </div>
    <div
      className="box"
    >
      <div
        className="input-group"
      >
        <label
          htmlFor="username"
        >
          Username
        </label>
        <input
          className="login-input"
          name="username"
          onChange={[Function]}
          placeholder="Username"
          type="text"
          value=""
        />
      </div>
      <div
        className="input-group"
      >
        <label
          htmlFor="password"
        >
          Password
        </label>
        <input
          className="login-input"
          name="password"
          onChange={[Function]}
          placeholder="Password"
          type="password"
          value=""
        />
      </div>
      <button
        className="login-btn"
        onClick={[Function]}
        type="button"
      >
        Login
      </button>
    </div>
  </div>
</div>
`;
